### DISCUSSION TITLE
Git LFS Migration for existing repository

### DISCUSSION THREAD

**Discussion by mano-si** [git-lfs/git-lfs#4487] [Q&A]
Hi! I have an existing  repo with .pbix files which are quite large in size. I wanted to migrate and track these files as LFS but had couple of questions. 
1. If I only do  git lfs track on a feature branch and merge the changes and .gitattribute to master, would that be of any help in reducing or maintaining the repo size in future. I do not do any history rewrite here. 
2. I do git lfs migrate import --include="*.pbix" --everything as suggested in documentation and push to master. Can this be done on a feature branch and merged to master? Do I need to do a git lfs track first before executing lfs migrate. 

Thank you!

**Comment by chrisd8088** [comment]
I recently rewrote the [man page](https://github.com/git-lfs/git-lfs/blob/94245ec834c7f938ec3a7808f73e9bd1d2b04f0a/docs/man/git-lfs-migrate.1.ronn) for `git-lfs-migrate(1)` in a way that I hope clarifies some of these kinds of questions.  Please do take a look, and suggest other changes if you have them.

Trying to answer to your specific questions, I would say:
1. Merging a branch as you describe would be no different than just running `git lfs track "*.pbix"` on your main branch.  It would add a `.gitattributes` file with `"*.pbix" filter=lfs diff=lfs merge=lfs -text` but would not automatically go back and change anything about your repo.  Future commits with `*.pbix` files would cause the Git LFS clean filter to run and convert them to LFS-managed objects, but existing `*.bpix` files would not change, unless you modified them and then committed that change.  If you happen to delete one of these older files in your work tree, and then try to restore it with, say, `git checkout -- .`, you'll get an `"Encountered 1 file(s) that should have been pointers, but weren't"` message.  So I would not recommend this approach as you're not migrating anything and going to cause problems managing files since only some will have been converted.
2. If you read through the revised man page, I hope it is explained better that `--everything` specifically means "all branches".  So it's going to churn through *all* branches, looking at all commits in your history on each branch; it doesn't matter whether you're on a feature branch or not.  Your whole local history will be rewritten (new commit SHAs) and will no longer just push to a remote without conflicts.  That's why the revised docs try to make it clear that after this command, you will need to force-push all branches to your remote, and that will ("destructively") change *its* entire history too.  That may well be what you want to do, in fact.  Just be clear in your mind what is happening.  I would suggest working locally with these commands, maybe on a copy of your repo or a test repo, until you fully understand their impact and feel comfortable with them.

**Reply by mano-si** [reply]
Thanks Chris! We did migrate with point 2 like you suggested and it has worked well. We went from over 2.5 GB to 450 MB . Thanks again! Would love to contribute to this awesome project in future.
